OutSystems/outsystems-maps
# null check on disposing uninitialised properties

Occasionally, when the map is hidden, users receive an error. This is due to is being disposed before it is fully initialised.

### What was happening
* If the maps component is disposed before initialisation is complete, throws an unhandled exception:
```
Cannot read properties of undefined (reading 'dispose') 
TypeError: Cannot read properties of undefined (reading 'dispose')
    at Map.dispose (https://mysite/MapsDisposeIssue/scripts/OutSystemsMaps.OutSystemsMaps.js?BAGpbsEzTulfFjBgJ28Zkg:2899:32)
    at Object.RemoveMap (https://mysite/MapsDisposeIssue/scripts/OutSystemsMaps.OutSystemsMaps.js?BAGpbsEzTulfFjBgJ28Zkg:7044:17)
    at https://mysite/MapsDisposeIssue/scripts/OutSystemsMaps.controller.js?wh8PgHyK8GB9Mo3ArBgS8Q:6347:19
    at e.safeExecuteJSNode (https://mysite/MapsDisposeIssue/scripts/OutSystems.js?eq9LGmzdgJbMq6dbNoVMvQ:16:7700)
    at OutSystemsMapsController.default.removeMap$Action (https://mysite/MapsDisposeIssue/scripts/OutSystemsMaps.controller.js?wh8PgHyK8GB9Mo3ArBgS8Q:6314:12)
    at Controller._onDestroy$Action (https://mysite/MapsDisposeIssue/scripts/OutSystemsMaps.Maps.Map.mvc.js?T3YTU+GttHWM5u1ffs_7Aw:568:34)
    at https://mysite/MapsDisposeIssue/scripts/OutSystems.js?eq9LGmzdgJbMq6dbNoVMvQ:16:18813
    at d (https://mysite/MapsDisposeIssue/scripts/OutSystems.js?eq9LGmzdgJbMq6dbNoVMvQ:2:16076)
    at f (https://mysite/MapsDisposeIssue/scripts/OutSystems.js?eq9LGmzdgJbMq6dbNoVMvQ:2:15822)
    at Object.p [as tryCatchFinally] (https://mysite/MapsDisposeIssue/scripts/OutSystems.js?eq9LGmzdgJbMq6dbNoVMvQ:2:16323)
```
See also, this discussion:
https://www.outsystems.com/forums/discussion/78788/cannot-read-properties-of-undefined-reading-dispose/

Note that this error is difficult to trigger as it requires a user to navigate away from the page or otherwise cause the block to be disposed while it is still loading.

### What was done
* Added null checks on dispose

### Test Steps
Note: this example is quite contrived, as the original issue is intermittent and depends on user timing. Here we dispose the map IMMEDIATELY, which would be odd in a real program, but serves to highlight the issue.

1. See MapsDisposeIssue.oml

or
1. Create a block containing a map. 
2. Wrap the block in an If block, with some trigger (eg button) to show the block.
3. Fire a Ready event in the block OnReady handler
4. In the parent, when handling the Ready event, hide the map

Actual Result: The Map's 'Destroy' will fire before it is finished initialising, triggering the error shown in the stracktrace.


### Screenshots
![MapDisposeIssue](https://user-images.githubusercontent.com/951052/171779088-e3bca0e2-7a1b-466c-ac84-38ea50c7563d.gif)


### Checklist
